src/metabase/xrays/automagic_dashboards/comparison.clj

```diff
@@ -248,67 +246,68 @@
-                                                       :model/Table
-                                                       (tru "All {0}" (:short-name right))
-
-                                                       (tru "{0}, all {1}"
-                                                            (comparison-name right)
-                                                            (names/source-name right)))))
-        segment-dashboards (->> (concat (segment-constituents left)
-                                        (segment-constituents right))
-                                distinct
-                                (map #(automagic-analysis % {:source       (:source left)
-                                                             :rules-prefix ["comparison"]})))]
-    (assert (or (let [left-source  (m/update-existing (:source left) :dataset_query magic.util/do-with-legacy-query identity)
-                      right-source (m/update-existing (:source right) :dataset_query magic.util/do-with-legacy-query identity)]
-                  (= left-source right-source))
-                (= (-> left :source :table_id)
-                   (-> right :source u/the-id))))
-    (->> (concat segment-dashboards [dashboard])
-         (reduce (fn [dashboard-1 dashboard-2]
-                   (if dashboard-1
-                     (populate/merge-dashboards dashboard-1 dashboard-2 {:skip-titles? true})
-                     dashboard-2))
-                 nil)
-         dashboard->cards
-         (m/distinct-by (some-fn :dataset_query hash))
-         (transduce (mapcat unroll-multiseries)
-                    (fn
-                      ([]
-                       (let [title (tru "Comparison of {0} and {1}"
-                                        (comparison-name left)
-                                        (comparison-name right))]
-                         (-> {:name              title
-                              :transient_name    title
-                              :transient_filters nil
-                              :param_fields      nil
-                              :description       (tru "Automatically generated comparison dashboard comparing {0} and {1}"
-                                                      (comparison-name left)
-                                                      (comparison-name right))
-                              :creator_id        api/*current-user-id*
-                              :parameters        []
-                              :related           (update-related (:related dashboard) left right)}
-                             (add-title-row left right))))
-                      ([[dashboard _row]] dashboard)
-                      ([[dashboard row] card]
-                       [(comparison-row dashboard row left right card)
-                        (+ row (:height card))]))))))
+  (binding [lib.schema/*HACK-disable-ref-validation* true]
```

Comment:
Can you add a comment to explain why this is necessary? (And ideally, under what conditions we could remove this hack?)